download_posts.py

```diff
@@ -1,25 +1,28 @@
 #!/usr/bin/python3
 
 import json
-import os 
-import requests 
+import os
+import requests
 from sys import exit as sysexit
 import xml.etree.ElementTree as ET
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
 
-# At time of script import, request range of years. We'll add 1 to the end year automatically.
-try:
-    startYear = int(input("Enter first year you want to export: "))
-except Exception as e:
-    print(f"\nError with first year entered. Error: {e}. Exiting...")
-    sysexit(1)
+DATE_FORMAT = '%Y%m%d'
 
-try:
-    endYear = int(input("Enter last year you want to export: "))
-except Exception as e:
-    print(f"\nError with last year entered. Error: {e}. Exiting...")
-    sysexit(1)
+def get_date_input(prompt):
+    try:
+        start_date = int(input("Enter start date in YYYYMMDD format: "))
```

Comment:
YYYYMMDD is not really a human-readable format. ISO8601 format YYYY-MM-DD would be better, or if we drop the requirement of the day, YYYY-MM would be enough. If this doesn’t work right away, that would be because of now unnecessary parsing into integer with `int()`.